subsystems/visionsubsystem.py

```diff
@@ -117,6 +119,106 @@ def reset(self) -> None:
         pass
 
 
+class BallTrackingModule:
+    def __init__(self) -> None:
+        self.targetAngle = None
+        self.targetDistance = None
+
+    def getTargetAngle(self) -> typing.Optional[Rotation2d]:
+        return self.targetAngle
+
+    def getTargetDistance(self) -> typing.Optional[float]:
+        return self.targetDistance
+
+    def update(self) -> None:
+        NetworkTables.initialize()
+        self.photonvisionNetworkTable = NetworkTables.getTable(
+            constants.kPhotonvisionNetworkTableName
+        )
+
+        if self.photonvisionNetworkTable.getBoolean(
+            constants.kPhotonvisionTargetValidKey, False
+        ):
+            if RobotBase.isReal():
```

Comment:
your general implementation of the system should NOT rely on if the robot is real - simulation should handle making the correct values